### PR TITLE
Account part of the database

### PR DESCRIPTION
**Pull request by clenrock**
- Relevant Issues: #27 
- Related Pull Requests: #57 

* * *

## What
It adds operations to add, delete, update and retrieve the accounts list from the database.

## Why
We need a part of the database which saves account information.

## How
To test this, it's possible to run the DatabaseTest.

## Notes
IMPORTANT: this branch merged the branch for pull request #57 in. So that PR needs to be merged first, before this one can be merged.
Also, the database now uses an array of Accounts, so that in the future we can have multiple accounts/profiles/keys :wink:

### CODE REVIEW

**Comment by ericcornelissen** [comment]
Just a quick first comment: I still feel like we need to go with a different name then "Accounts", as discussed in #50

File: app/src/test/java/com/nervousfish/nervousfish/modules/database/DatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    @Test
+    public void testGetAccountEmpty() throws IOException {
+        List<Account> actual = database.getAccounts();
+        assertEquals(new ArrayList<Account>(), actual);
+    }
+
+    @Test
+    public void testAddAccount() throws IOException {
```
Comment (TheBonheurs) [inline app/src/test/java/com/nervousfish/nervousfish/modules/database/DatabaseTest.java]: I really like how you test account adding/deleting here 👍 . I think this should also be done with contacts in #57

**Comment by clenrock** [comment]
@ericcornelissen I agree, but are we going to change that now, or the next sprint? (where we will discuss this in person)

**Comment by ericcornelissen** [comment]
> I agree, but are we going to change that now, or the next sprint? (where we will discuss this in person)

I'm not going to accept this PR before we figured it out, I personally don't care too much whether that is this or next sprint 😉

**Comment by clenrock** [comment]
@ericcornelissen What do you suggest as a fix?

File: app/src/test/java/com/nervousfish/nervousfish/modules/database/DatabaseTest.java (hunk 1 above)
Comment (ericcornelissen) [inline app/src/test/java/com/nervousfish/nervousfish/modules/database/DatabaseTest.java]: I'm not sure if I agree. I think that, in unittest, you want to accomplish that only one part of the functionality is tested so you can easily trace back problems. What you're doing in this test is testing both the functionality of `addAccount` and `getAccount`...

@jverbraeken, what is your opinion in this?

**Comment by ericcornelissen** [comment]
> @ericornelissen What do you suggest as a fix?

I've stated my opinion in issue #50. I personally prefer the term "Profile" but I'm open to other suggestions 🙂

**Comment by clenrock** [comment]
@ericcornelissen I don't see much difference in the name Account or Profile, I think they mean the same thing.

File: app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java [hunk 2]
```diff
@@ -32,14 +35,17 @@
 /**
  * An adapter to the GSON database library
  */
+@SuppressWarnings("PMD.TooManyMethods")
 public final class GsonDatabaseAdapter implements IDatabase {
 
     private final static Type TYPE_CONTACT_LIST = new TypeToken<ArrayList<Contact>>(){}.getType();
-
+    private final static Type TYPE_PROFILE_INFORMATION = new TypeToken<ArrayList<Profile>>(){}.getType();
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java]: @clenrock, all caps :wink:

**Pull request merged**